### PR TITLE
2 api documentation architecture presentation asif

### LINKED ISSUE #2: API Documentation & Architecture presentation
1. Design the Architecture
2. Write the mock API documentation

### PR DESCRIPTION
**Pull request by asifhaider**
@ANIK115 do a review of my changes and merge if everything is alright.

### CODE REVIEW

File: API-Doc.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+| --- | --- | --- | --- | --- |
+| ```GET``` Professor Suggestion | | | | |
+| [beyond-the-seas.org/api/professor]() | GET | ```{ "area_of_interest": ["Big Data Analytics", "Data Science"] }``` | 200 | ```{ "professors": [] }``` |
+| ```GET``` Search Professor | | | | |
+| [beyond-the-seas.org/api/professor/search?start=0&count=10]() | GET | ```{ "keyword": "Texas" }``` | 200 | ```{ "professors": [] }``` |
+| ```GET``` Professor Details | | | | |
+| [beyond-the-seas.org/api/professor/{professor_id}/details]() | GET | ```{ "professor_id": 3 }``` | 200 | ```{ "professor": {"name": "Latifur Khan", "designation": "Professor", "department": "Computer Science", "university": "University of Texas at Dallas", "research-area": ["Big Data", "Data Streams", "Cyber Security"], "email": "khan@utdallas.edu", "address": "3322 ECSS, Dallas, TX, USA", "website": "https://cs.utdallas.edu/people/faculty/khan-latifur/", "scholar-info": ["14004", "65"], "funding-opportunity": [], "ongoing-projects": [], "feedback": []}``` |
+| ```GET``` Provide Analytics Data to Analytics Service | | | | |
+| [beyond-the-seas.org/api/professor/analytics]() | GET | ```{}``` | 200 | ```{ "analytics": {} }``` |
```
Comment (ANIK115) [inline API-Doc.md:46]: The request body should have a user_id

**Review by ANIK115 — CHANGES_REQUESTED** [review]

File: API-Doc.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
```
Comment (ANIK115) [inline API-Doc.md:48]: You need to add an api for student's profile matching with professor's requirements

**Review by ANIK115 — CHANGES_REQUESTED** [review]

File: API-Doc.md (hunk 1 above)
Comment (asifhaider) [reply API-Doc.md:46]: Fixed

File: API-Doc.md (hunk 2 above)
Comment (asifhaider) [reply API-Doc.md:48]: Done

**Pull request merged**